packages-ts/gauntlet-terra-contracts/src/commands/abstract/executionWrapper.ts

```diff
@@ -21,7 +28,7 @@ export interface AbstractInstruction<Input, ContractInput> {
   makeInput: (flags: any, args: string[]) => Promise<Input>
   validateInput: (input: Input) => boolean
   makeContractInput: (input: Input) => Promise<ContractInput>
-  beforeExecute?: (context: BeforeExecutionContext<Input, ContractInput>) => () => Promise<void>
+  beforeExecute?: BeforeExecute<Input, ContractInput>
   afterExecute?: (response: Result<TransactionResponse>) => any
```

Comment:
Feel free to also type this function, something like:

```
export type AfterExecute<Input, ContractInput> = (
  context: ExecutionContext<Input, ContractInput>,
  response: Result<TransactionResponse>,
) => Promise<any>
```

Additionally, not too fond of these `<Input, ContractInput>` types overall but no suggestions yet.